returnn/datasets/distrib_files.py

```diff
@@ -245,8 +264,32 @@ def init_seq_order(self, epoch: Optional[int] = None, seq_list=None, seq_order=N
         self._lazy_init_file_sizes()
         self._lazy_init_file_cache_proc()
 
-        # Cleanup and fill _files_order_cache.
         full_epoch_0idx = (epoch - 1) // self.partition_epoch
+
+        # Shard files across GPU workers
+        for k in list(self._files_for_this_worker.keys()):
+            if k < full_epoch_0idx:
+                del self._files_for_this_worker[k]
+        for ep_ in range(epoch, epoch + self.preload_next_n_sub_epochs + 1):
+            full_epoch_0idx_ = (ep_ - 1) // self.partition_epoch
+            if full_epoch_0idx_ in self._files_for_this_worker:
+                continue
+            if self.seq_ordering == "default":
+                files = self.files
+            elif self.seq_ordering == "random":
+                # do not use `self._get_random_seed_for_epoch` to exclude `random_seed_offset`
+                # to ensure an equal distribution in every worker
```

Comment:
```suggestion
                # to ensure an equal distribution in every worker
                # since the sequences might collide between workers if a different seed is chosen.
```